File: frontend-app/profile-page/src/model/VehicleResponseDTO.ts
import { VehicleStatusResponseDTO } from './VehicleStatusResponseDTO';
import { VehicleDetailsDTO } from './VehicleDetailsDTO';
import { EquipmentResponseDTO } from './EquipmentResponseDTO';
import {LocationResponseDTO} from "./LocationResponseDTO";

export class VehicleResponseDTO {
    id: string;
    registration: string;
    brand: string;
    model: string;
    dailyFee: number;
    location: LocationResponseDTO;
    bestOffer: boolean;
    vehicleStatus: VehicleStatusResponseDTO;
    vehicleDetails: VehicleDetailsDTO;
    equipments: EquipmentResponseDTO[];

    constructor(
        id: string,
        brand: string,
        model: string,
        dailyFee: number,
        registration: string,
        location: LocationResponseDTO,
        bestOffer: boolean,
        vehicleStatus: VehicleStatusResponseDTO,
        vehicleDetails: VehicleDetailsDTO,
        equipments: EquipmentResponseDTO[]
    ) {
        this.id = id;
        this.brand = brand;
        this.model = model;
        this.dailyFee = dailyFee;
        this.registration = registration;
        this.location = location;
        this.bestOffer = bestOffer;
        this.vehicleStatus = vehicleStatus;
        this.vehicleDetails = vehicleDetails;
        this.equipments = equipments;
    }
}

File: api/vehicle-service/src/main/java/com/carrental/vehicleservice/model/dto/VehicleResponseDTO.java
package com.carrental.vehicleservice.model.dto;

import lombok.Data;

import java.math.BigDecimal;
import java.util.Set;

@Data
public class VehicleResponseDTO {

    private Long id;

    private String registration;

    private String brand;

    private String model;

    private BigDecimal dailyFee;

    private LocationResponseDTO location;

    private boolean bestOffer;

    private VehicleStatusResponseDTO vehicleStatus;

    private VehicleDetailsDTO vehicleDetails;

    private Set<EquipmentResponseDTO> equipments;

    private Double averageRate;
}
